server/fleet/app_test.go

```diff
@@ -0,0 +1,169 @@
+					StatusStream: config.TestFirehosePluginConfig().Firehose.StatusStream,
+					ResultStream: config.TestFirehosePluginConfig().Firehose.ResultStream,
+				},
+				Kinesis: nil,
+				Lambda:  nil,
+				PubSub:  nil,
+			},
+		},
+		{
+			name: "Lambda Logging Plugin serializes Properly",
+			args: args{conf: config.TestLambdaPluginConfig()},
+			want: &Logging{
+				Debug:           true,
+				Json:            false,
+				ResultLogPlugin: "lambda",
+				StatusLogPlugin: "lambda",
+				FileSystem:      nil,
+				Firehose:        nil,
+				Kinesis:         nil,
+				Lambda: &LambdaConfig{
+					Region:         config.TestLambdaPluginConfig().Lambda.Region,
+					StatusFunction: config.TestLambdaPluginConfig().Lambda.StatusFunction,
+					ResultFunction: config.TestLambdaPluginConfig().Lambda.ResultFunction,
+				},
+				PubSub: nil,
+			},
+		},
+		{
+			name: "PubSub Logging Plugin serializes Properly",
+			args: args{conf: config.TestPubSubPluginConfig()},
+			want: &Logging{
+				Debug:           true,
+				Json:            false,
+				ResultLogPlugin: "pubsub",
+				StatusLogPlugin: "pubsub",
+				FileSystem:      nil,
+				Firehose:        nil,
+				Kinesis:         nil,
+				Lambda:          nil,
+				PubSub: &PubSubConfig{config.PubSubConfig{
+					Project:       config.TestPubSubPluginConfig().PubSub.Project,
+					StatusTopic:   config.TestPubSubPluginConfig().PubSub.StatusTopic,
+					ResultTopic:   config.TestPubSubPluginConfig().PubSub.ResultTopic,
+					AddAttributes: config.TestPubSubPluginConfig().PubSub.AddAttributes,
+				}},
+			},
+		},
+		{
+			name: "Stdout Logging Plugin serializes Properly",
+			args: args{conf: config.FleetConfig{
+				Osquery:    config.OsqueryConfig{ResultLogPlugin: "stdout", StatusLogPlugin: "stdout"},
+			}},
+			want: &Logging{
+				Debug:           false,
+				Json:            false,
+				ResultLogPlugin: "stdout",
+				StatusLogPlugin: "stdout",
+				FileSystem:      nil,
+				Firehose:        nil,
+				Kinesis:         nil,
+				Lambda:          nil,
+				PubSub:          nil,
+			},
+		},
+		{
+			name: "Empty Logging Plugin serializes Properly",
+			args: args{conf: config.FleetConfig{
+				Osquery:    config.OsqueryConfig{ResultLogPlugin: "", StatusLogPlugin: ""},
+			}},
+			want: &Logging{
+				Debug:           false,
+				Json:            false,
+				ResultLogPlugin: "filesystem",
+				StatusLogPlugin: "filesystem",
+				FileSystem:      &FileSystemConfig{config.FilesystemConfig{
+					StatusLogFile:        "",
+					ResultLogFile:        "",
+					EnableLogRotation:    false,
+					EnableLogCompression: false,
+				}},
+				Firehose:        nil,
+				Kinesis:         nil,
+				Lambda:          nil,
+				PubSub:          nil,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := LoggingFromConfig(tt.args.conf); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("LoggingFromConfig() = %v, want %v", got, tt.want)
+			}
```

Comment:
Please prefer `assert.Equal` and other helpers from `github.com/testify/assert` for tests.